lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -176,13 +176,40 @@ def credit(amount, payment, options = {})
         end
       end
 
-      def verify(credit_card, options = {})
+      def verify(payment_method, options = {})
+        amount = define_amount_for_verify(payment_method, options)
+        return amount if amount.class == Response
+
         MultiResponse.run(:use_first_response) do |r|
-          r.process { authorize(100, credit_card, options) }
-          r.process(:ignore_result) { void(r.authorization, options) }
+          r.process { authorize(amount, payment_method, options) }
+          r.process(:ignore_result) { void(r.authorization, options) } unless amount == 0
         end
       end
 
+      def define_amount_for_verify(payment_method, options)
+        if options[:verify_amount].present?
+          if options[:verify_amount].class != Integer || options[:verify_amount] < 0
+            response = Response.new(false, 'verify_amount value must be an integer')
+          elsif options[:verify_amount] > 0
+            response = options[:verify_amount]
+          elsif options[:verify_amount] == 0
+            response = 0
+            response = Response.new(false, 'zip and address must be different than blank value') unless validate_billing_address_values?(options) 
```

Comment:
Have we been able to confirm that these are the fields that are actually required for a $0 auth?

Reply:
After several test with the combinations we have found these couple as the minimum requirement to 0 auth